### PR TITLE
[Fortune] Lower case addresses in signatures

### PR DESCRIPTION
**Pull request by eugenvoronov**
## Description
In the signature verification and sign message functions if the message contains a an address convert it to lower case.

## Summary of changes
Implemented `toLowerCase`.

## How test the changes
`yarn test`

## Related issues
[Fortune Exchange Oracle] Lower case addresses in signatures
[#2206](https://github.com/humanprotocol/human-protocol/issues/2206)
[Fortune Recording Oracle] Lower case addresses in signatures
[#2207](https://github.com/humanprotocol/human-protocol/issues/2207)

### CODE REVIEW

File: packages/apps/fortune/exchange-oracle/server/src/common/utils/index.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export function toLowerCase(input: string | undefined): string {
```
Comment (portuu3) [inline packages/apps/fortune/exchange-oracle/server/src/common/utils/index.ts:1]: This doesn't make sense, just use native toLowerCase

**Review by portuu3 — CHANGES_REQUESTED** [review]

File: packages/apps/fortune/exchange-oracle/server/src/common/guards/signature.auth.ts [hunk 2]
```diff
@@ -27,7 +27,13 @@ export class SignatureAuthGuard implements CanActivate {
     );
     if (!roles) throw new NotImplementedException(ErrorSignature.MissingRoles);
     const request = context.switchToHttp().getRequest();
-    const data = request.body;
+    const { escrow_address, ...restBody } = request.body;
```
Comment (portuu3) [inline packages/apps/fortune/exchange-oracle/server/src/common/guards/signature.auth.ts:30]: We should the lower case conversion in the signature verification (method verifySignature). If we do it here and we call verifySignature from somewhere else the issue will persist. So this changes here are not required

File: packages/apps/fortune/recording-oracle/src/modules/job/job.service.spec.ts [hunk 3]
```diff
@@ -136,7 +136,7 @@ describe('JobService', () => {
       (EscrowClient.build as jest.Mock).mockResolvedValue(escrowClient);
 
       const jobSolution: WebhookDto = {
-        escrowAddress: MOCK_ADDRESS,
+        escrowAddress: MOCK_ADDRESS.toLowerCase(),
```
Comment (portuu3) [inline packages/apps/fortune/recording-oracle/src/modules/job/job.service.spec.ts:139]: If the implementation is correct we should not requiered to pass lowercase in the tests, in should be done internally.

**Review by portuu3 — CHANGES_REQUESTED** [review]

**Comment by Dzeranov** [comment]
@eugenvoronov @portuu3 
Why can't we just modify a bit the structure here?
As far as I can see, we're actually verifying a specific data structure:
```
{
  chain_id: ChainId,
  escrow_address: string,
  event_type: EventType,
  event_data?: EventData
}
```
So we actually don't need to apply any `.stringify()` before data comes to `.verifySignature()`.
In this case we can specify an exact type of data/message and make all the necessary case conversions inside e.g.

```
export function verifySignature(
  data: WebhookDto,
  signature: string,
  addresses: string[],
): boolean {
  // set escrow_address to lower case
  // stringify a message
  const signer = recoverSigner(message, signature);

  ...
  }

  return true;
}
```

**Review by portuu3 — DISMISSED** [review]

**Review by portuu3 — APPROVED** [review]

**Pull request merged**